tck/api/src/main/java/org/eclipse/microprofile/metrics/tck/GlobalTagsTest.java

```diff
@@ -0,0 +1,102 @@
+ *               2010-2013 Coda Hale, Yammer.com
+ *
+ * See the NOTICES file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+
+package org.eclipse.microprofile.metrics.tck;
+
+import org.eclipse.microprofile.metrics.MetricFilter;
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.Tag;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.inject.Inject;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+/**
+ * Test for obtaining global tags from MP_METRICS_TAGS env variable or mp.metrics.tags system property.
+ */
+@RunWith(Arquillian.class)
+public class GlobalTagsTest {
+
+    @Inject
+    private MetricRegistry registry;
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class).addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @After
+    public void cleanupApplicationMetrics() {
+        registry.removeMatching(MetricFilter.ALL);
+    }
+
+    /**
+     * This expects that there is a MP_METRICS_TAGS env variable with value 'tier=integration', as described in running_the_tck.asciidoc
+     */
+    @Test
+    public void fromEnvVariable() {
+        registry.counter("mycounter", new Tag("foo", "bar"));
+        final MetricID actualMetricId = registry.getCounters().keySet().stream().filter(id -> id.getName().equals("mycounter")).findAny().get();
+        Assert.assertThat(actualMetricId.getTagsAsList(), containsInAnyOrder(
+            new Tag("foo", "bar"),
+            new Tag("tier", "integration")
+        ));
+    }
+
+    /**
+     * Setting a system property mp.metrics.tags should override the MP_METRICS_TAGS env variable.
+     */
+    @Test
+    public void fromSystemProperty() {
+        String previousValue = System.setProperty("mp.metrics.tags", "key1=value1,key2=value2");
```

Comment:
In mpConfig 1.3 (and below), the System Property Config Source is static and wouldn't be able to change "dynamically" during runtime.

Reply:
Hmmmmm... From 1.3 specification: https://github.com/eclipse/microprofile-config/blob/1.3/spec/src/main/asciidoc/configsources.asciidoc#configsource-and-mutable-data

```
A Config instance provides no caching but iterates over all ConfigSources for each getValue(String) operation. A ConfigSource is allowed to cache the underlying values itself.
```

This implies to me that one ConfigSource is allowed to cache the config values, but we're using two different config sources:
- first test is using environment config source
- second test is using system property config source 

So when the second test sets a system property, then because a system property config source has higher priority, MP Config should pick it up and it should work. 
Yeah this will only work if the test for the env variable comes first, because in the other case the value from the system property might get cached and be returned in the ENV variable test, even after we clear the system property. Which would mean we should add `@InSequence` annotations to force the order of the tests.

But even with that, I guess that it might interfere with other test cases that come in later, because the value from the system property could persist within the scope of the same JVM. So yeah this gets problematic :( to be clean we would have to make sure the test with the system property runs with its own separate container/JVM, which would probably be too disruptive to be worth doing that. So unless we come up with a different clean way, I guess we have to get rid of the test `fromSystemProperty`...